plugins/explore-backend/src/tools/types.ts

```diff
@@ -31,3 +32,17 @@ export interface ExploreToolProvider {
    */
   getTools(request: GetExploreToolsRequest): Promise<GetExploreToolsResponse>;
 }
+
+/** @public */
+export interface ExploreProviderRegistrationOptions {
+  providerId: string;
+  factory: StaticExploreToolProvider;
+}
+
+/** @public */
+export interface ExporeProvidersExtensionPoint {
+  registerProvider(options: ExploreProviderRegistrationOptions): void;
```

Comment:
Prefer `setToolProvider`, and it shouldn't need an id, and no need for it to be a factory either, just the instance (the interface type for those is `ExploreToolProvider`, not the implementation class `StaticExploreToolProvider`).

And it's just a setter that should only be allowed to be called once, trying to register multiple competing providers should be an error